internal/scheduler/constraints/constraints.go

```diff
@@ -0,0 +1,167 @@
+	}
+
+	// MaximalResourceFractionToSchedule check.
+	totalScheduledResources := sctx.ScheduledResourcesByPriority.AggregateByResource()
+	if exceeded, reason := exceedsResourceLimits(
+		ctx,
+		totalScheduledResources,
+		sctx.TotalResources,
+		constraints.MaximalResourceFractionToSchedule,
+	); exceeded {
+		unschedulableReason := reason + " (overall per scheduling round limit)"
+		return false, unschedulableReason, nil
+	}
+
+	return true, "", nil
+}
+
+func (constraints *SchedulingConstraints) CheckPerQueueAndPriorityClassConstraints(
+	ctx context.Context,
+	sctx *schedulercontext.SchedulingContext,
+	queue string,
+	priorityClassName string,
+) (bool, string, error) {
+	// We assume that all jobs in a gang have the same priority class (which we enforce at job submission).
+	qctx := sctx.QueueSchedulingContexts[queue]
+	if qctx == nil {
+		return false, "", errors.Errorf("no QueueSchedulingContext for queue %s", queue)
+	}
+
+	// MaximalResourceFractionToSchedulePerQueue check.
+	if exceeded, reason := exceedsResourceLimits(
+		ctx,
+		qctx.ScheduledResourcesByPriority.AggregateByResource(),
+		sctx.TotalResources,
+		constraints.MaximalResourceFractionToSchedulePerQueue,
+	); exceeded {
+		unschedulableReason := reason + " (per scheduling round limit for this queue)"
+		return false, unschedulableReason, nil
+	}
+
+	// MaximalResourceFractionPerQueue check.
+	if exceeded, reason := exceedsResourceLimits(
+		ctx,
+		qctx.AllocatedByPriority.AggregateByResource(),
+		sctx.TotalResources,
+		constraints.MaximalResourceFractionPerQueue,
+	); exceeded {
+		unschedulableReason := reason + " (total limit for this queue)"
+		return false, unschedulableReason, nil
+	}
+
+	// PriorityClassSchedulingConstraintsByPriorityClassName check.
+	if priorityClassConstraint, ok := constraints.PriorityClassSchedulingConstraintsByPriorityClassName[priorityClassName]; ok {
+		allocatedByPriorityAndResourceType := schedulerobjects.NewAllocatedByPriorityAndResourceType([]int32{priorityClassConstraint.PriorityClassPriority})
+		for p, rl := range qctx.AllocatedByPriority {
+			allocatedByPriorityAndResourceType.MarkAllocated(p, rl)
+		}
+		if exceeded, reason := exceedsResourceLimits(
+			ctx,
+			schedulerobjects.QuantityByPriorityAndResourceType(allocatedByPriorityAndResourceType).AggregateByResource(),
+			sctx.TotalResources,
+			priorityClassConstraint.MaximumCumulativeResourceFractionPerQueue,
+		); exceeded {
+			unschedulableReason := reason + fmt.Sprintf(" for priority class %s (total limit for this queue)", priorityClassName)
+			return false, unschedulableReason, nil
+		}
+	}
+
+	return true, "", nil
+}
+
+// exceedsResourceLimits returns true if used/total > limits for some resource t,
+// and, if that is the case, a string indicating which resource limit was exceeded.
+func exceedsResourceLimits(_ context.Context, used, total schedulerobjects.ResourceList, limits map[string]float64) (bool, string) {
```

Comment:
why are we passing the context through here if it isn't used?

Reply:
Removed the conttext.